packages/frontend-2/components/settings/Dialog.vue

```diff
@@ -102,37 +89,39 @@
 </template>
 
 <script setup lang="ts">
-import type { defineComponent } from 'vue'
-import SettingsUserProfile from '~/components/settings/user/Profile.vue'
-import SettingsUserNotifications from '~/components/settings/user/Notifications.vue'
-import SettingsUserDeveloper from '~/components/settings/user/Developer.vue'
-import SettingsServerGeneral from '~/components/settings/server/General.vue'
-import SettingsServerProjects from '~/components/settings/server/Projects.vue'
-import SettingsServerActiveUsers from '~/components/settings/server/ActiveUsers.vue'
-import SettingsServerPendingInvitations from '~/components/settings/server/PendingInvitations.vue'
-import SettingsWorkspacesMembers from '~/components/settings/workspaces/Members.vue'
+import type { SettingsMenuItem } from '~/lib/settings/helpers/types'
+import { useIsWorkspacesEnabled } from '~/composables/globals'
+import { useQuery } from '@vue/apollo-composable'
+import { settingsSidebarWorkspacesQuery } from '~/lib/settings/graphql/queries'
 import { useBreakpoints } from '@vueuse/core'
 import { TailwindBreakpoints } from '~~/lib/common/helpers/tailwind'
 import { UserIcon, ServerStackIcon } from '@heroicons/vue/24/outline'
-import { settingsQueries } from '~/lib/common/helpers/route'
 import { useActiveUser } from '~/lib/auth/composables/activeUser'
+import { useSettingsMenu } from '~/lib/settings/composables/menu'
 import {
   LayoutSidebar,
   LayoutSidebarMenu,
   LayoutSidebarMenuGroup
 } from '@speckle/ui-components'
 import { Roles } from '@speckle/shared'
-import { useQuery } from '@vue/apollo-composable'
-import { settingsSidebarWorkspacesQuery } from '~/lib/settings/graphql/queries'
-import { useIsWorkspacesEnabled } from '~/composables/globals'
+import { graphql } from '~~/lib/common/generated/gql'
 
-type MenuItem = {
-  title: string
-  component: ReturnType<typeof defineComponent>
-}
+graphql(`
+  fragment SettingsSidebarWorkspaces_WorkspaceCollection on WorkspaceCollection {
```

Comment:
1. The name of the component is: SettingsDialog, so the fragment name is off
2. I think it would be better if the fragment were defined on `User` instead (Ask for `workspaces`), cause otherwise it feels like you're omitting a part of the requirement. Considering that this is the only fragment the query asks for, its a bit confusing that the fragment only paints part of the picture (asking for some kind of workspace collection) while the query then fills out the rest (specifying that its supposed to be the active user's workspace collection)